arelle/ModelValue.py

```diff
@@ -919,3 +922,30 @@ def __new__(cls, value: str) -> InvalidValue:
         return str.__new__(cls, value)
 
 INVALIDixVALUE = InvalidValue("(ixTransformValueError)")
+
+TypeSValue = Union[
+    bool,
+    float,
+    int,
+    InvalidValue,
+    None,
+    str,
+]
+TypeXValue = Union[
```

Comment:
We would also want to add `XsdPattern` here. I'm not sure how though as it creates a circular import.